### ISSUE TITLE
AWS::Serverless::Function.EventInvokeConfig.MaximumRetryAttempts constraints not enforced

### ISSUE THREAD

**Issue by farski** [aws-cloudformation/cfn-lint#1926]
*cfn-lint version: 0.46.0*

*Description of issue.*

```yaml
  RelayFunction:
    Type: AWS::Serverless::Function
    Properties:
      EventInvokeConfig:
        MaximumRetryAttempts: 5
```

`cfn-lint` doesn't complain about this setup, but deploying it results in:

> 1 validation error detected: Value '5' at 'maximumRetryAttempts' failed to satisfy constraint: Member  must have value less than or equal to 2

**Comment by kddejong** [comment]
@farski I believe `MaximumRetryAttempts` is converted to `AWS::Lambda::EventSourceMapping` Looks like that [limit](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-eventsourcemapping.html#cfn-lambda-eventsourcemapping-maximumretryattempts) is pretty high so maybe the limit is conditional based on the event source its mapped to.  What is your event source?

**Comment by farski** [comment]
```yaml
      EventInvokeConfig:
        DestinationConfig:
          OnFailure:
            Destination: !Ref RelayFunctionFailureSnsTopic
            Type: SNS
        MaximumRetryAttempts: 2
```

**Comment by kddejong** [comment]
According to the docs this value looks like it may only apply to streams.  Retries on asynchronous resources is set to 2.  Can you change that? Been a year since I really dove deep on this so I apologize if I'm a little bit behind on the details here.

```
MaximumRetryAttempts
    (Streams) Discard records after the specified number of retries. The default value is infinite (-1). When set to infinite (-1), failed records are retried until the record expires.
    Required: No
    Type: Integer
    Minimum: -1
    Maximum: 10000
    Update requires: No interruption
```

**Comment by kddejong** [comment]
V1 now has this error:
```
E3034 5 is greater than the maximum of 2
local/issue/1926.yaml:2:1
```
translated template resource

```json
"RelayFunctionEventInvokeConfig": {
   "Properties": {
    "DestinationConfig": {},
    "FunctionName": {
     "Ref": "RelayFunction"
    },
    "MaximumRetryAttempts": 5,
    "Qualifier": "$LATEST"
   },
   "Type": "AWS::Lambda::EventInvokeConfig"
  },
```
